### ISSUE TITLE
False positive on E1102 (not-callable) for methods called prev or next

### ISSUE THREAD

**Issue by mathieui** [PyCQA/pylint#4232]
### Steps to reproduce

([failing CI job on non-minimal code](https://lab.louiz.org/mathieui/poezio/-/jobs/31905#L2400))
It took me a while to isolate a minimal use case, from my repeated testing:

* There is a special case, in *any object*, for ``prev()`` and ``next()``
* It is only the case if the ``typing`` module is imported

Given a file `test.py`:
```python
import typing

class MyClass:
    def __init__(self):
        pass

    def next(self):
        pass

    def prev(self):
        pass

    def toto(self):
        pass

    def placeholder(self):
        self.prev()
        self.next()
        self.toto()
```

### Current behavior

Result of `pylint -E test.py`:
```
************* Module test
test.py:17:8: E1102: self.prev is not callable (not-callable)
test.py:18:8: E1102: self.next is not callable (not-callable)

```

### Expected behavior

No error.

### pylint --version output

Result of `pylint --version` output:
```
pylint 2.8.0-dev1
astroid 2.6.0-dev1
Python 3.9.2 (default, Feb 20 2021, 18:40:11) 
[GCC 10.2.0]
```

(installed both pylint and astroid from git to check)

**Comment by cdce8p** [comment]
This seems to be related to #3970 and possibly #3595

**Comment by mathieui** [comment]
Indeed. I was mostly focused on the ``prev()`` which did not appear in my search before submitting the issue, but it certainly looks like a duplicate of both.

**Comment by hippo91** [comment]
@mathieui i will close this issue as it seems to be a duplicate of at least #3970.
